staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -101,6 +110,8 @@ func newStore(c *clientv3.Client, pagingEnabled bool, codec runtime.Codec, prefi
 		pathPrefix:   path.Join("/", prefix),
 		watcher:      newWatcher(c, codec, versioner, transformer),
 		leaseManager: newDefaultLeaseManager(c),
+		// TODO(immutableT) Get this from the transformer.
+		listTransformationConcurrency: 4,
```

Comment:
- Computing the level of concurrency based on the timing of the first few calls sounds very interesting. So that I don't reinvent the wheel, are there any places where we do this already? 
I am not clear on how to translate the latency of a service to the level of concurrency it could handle.

- I agree (at least at the moment) it is only the KMS transformer that would benefit from concurrent processing. The [encryptionconfig/config](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go#L231) should be able to determine if any of the encryption providers (in the EncryptionConfig.yaml) would benefit from concurrent processing, and could pass this knowledge upstream to the storagefactory. Does this seam like a reasonable approach?

Reply:
I think the encryption provider should be required to tell us whether concurrency is needed.

Reply:
I agree that the encryption provider should be required to specify its concurrency level since we can't know how scalable the remote KMS is.

I think that an approach we could take is to extend the Transformer interface:
`TansformFromStorageConcurrencyLevel() int`

What do you think about how to best represent concurrency level within PrefixTransformer - this is what we pass into etcd3.store.
Since PrefixTransformer may wrap several underlying transformers (each with potentially different levels of concurrency), what would be a good way to handle this edge case (min or max come to mind)?